File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        set<char> st;
        int count=0;
        for(int i=0;i<s.length();i++){
            st.insert(s[i]);
        }
        for(auto it:st){
            int i=0,j=s.length()-1;
            while(s[i] != it){
                i++;
            }
            while(s[j] != it){
                j--;
            }
            set<char> temp;
            if(i<j-1){
            for(int k=i+1;k<j;k++){
                temp.insert(s[k]);
            }
            }
            count = count + temp.size();

        }
        return count;
    }
};

File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.py
class Solution(object):
    def countPalindromicSubsequence(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans = 0
        unique = set()
        for i in s:
            unique.add(i)
        
        for ch in unique:
            i=0
            j =len(s)-1
            while(s[i] != ch):
                i +=1
            while(s[j] != ch):
                j -=1
            temp = set()
            i +=1
            while(i<j):
                temp.add(s[i])
                i+=1
            ans +=len(temp)
        
        return ans
        